pkg/miniogw/gateway-storj.go

```diff
@@ -5,27 +5,196 @@ package storj
 	"time"
 
 	"github.com/minio/cli"
+	"github.com/vivint/infectious"
+	"storj.io/storj/pkg/eestream"
 
-	"github.com/minio/minio/cmd"
+	minio "github.com/minio/minio/cmd"
 	"github.com/minio/minio/pkg/auth"
 	"github.com/minio/minio/pkg/hash"
 )
 
+var (
+	pieceBlockSize = flag.Int("piece_block_size", 4*1024, "block size of pieces")
+	key            = flag.String("key", "a key", "the secret key")
+	rsk            = flag.Int("required", 20, "rs required")
+	rsn            = flag.Int("total", 40, "rs total")
+)
+
+// S3CliAPI contains parameters for accessing the Storj network
+type S3CliAPI struct {
+	totalBuckets int
+	bucketlist   []S3BucketList
+}
+
+//S3BucketList structure
+type S3BucketList struct {
+	bucket   minio.BucketInfo
+	filelist S3FileList
+}
+
+//S3FileList structure
+type S3FileList struct {
+	totalFiles int
+	file       minio.ListObjectsInfo
+}
+
+// gS3Gateway global S3 interface structure
+var gS3CliAPI S3CliAPI
+
+//createDummyBucketList function initializes sample buckets and files in each bucket
+func createDummyBucketList() {
+	gS3CliAPI.bucketlist = gS3CliAPI.bucketlist[:0]
+	for i := 0x00; i < 0x0A; i++ {
+		gS3CliAPI.bucketlist = append(gS3CliAPI.bucketlist,
+			S3BucketList{
+				bucket: minio.BucketInfo{
+					Name:    "TestBucket" + strconv.Itoa(i+1),
+					Created: time.Now(),
+				},
+			},
+		)
+	} /* end of for loop */
+
+	for i := 0x00; i < 0x0A; i++ {
+		gS3CliAPI.bucketlist[i].filelist.file.IsTruncated = false
+		gS3CliAPI.bucketlist[i].filelist.file.Objects = gS3CliAPI.bucketlist[i].filelist.file.Objects[:0]
+		for j := 0x00; j < 0x0A; j++ {
+			gS3CliAPI.bucketlist[i].filelist.file.Objects = append(
+				gS3CliAPI.bucketlist[i].filelist.file.Objects,
+				minio.ObjectInfo{
+					Bucket:      gS3CliAPI.bucketlist[i].bucket.Name,
+					Name:        gS3CliAPI.bucketlist[i].bucket.Name + "file" + strconv.Itoa(j+1),
+					ModTime:     time.Now(),
+					Size:        int64(100 + i),
+					ContentType: "application/octet-stream",
+				},
+			)
+		}
+	}
+	gS3CliAPI.totalBuckets = len(gS3CliAPI.bucketlist)
+	fmt.Println("bucket name = ", gS3CliAPI.bucketlist)
+}
+
 func init() {
-	cmd.RegisterGatewayCommand(cli.Command{
+	minio.RegisterGatewayCommand(cli.Command{
 		Name:            "storj",
 		Usage:           "Storj",
 		Action:          storjGatewayMain,
 		HideHelpCommand: true,
 	})
+
+	// create dummy bucket list
+	createDummyBucketList()
+}
+
+// getbucketscallback
+func getbucketscallback() (buckets []minio.BucketInfo, err error) {
+	buckets = make([]minio.BucketInfo, gS3CliAPI.totalBuckets)
+	for i, bi := range gS3CliAPI.bucketlist {
+		buckets[i] = minio.BucketInfo{
+			Name:    bi.bucket.Name,
+			Created: bi.bucket.Created,
+		}
+	}
+	fmt.Println("buckets list =", buckets)
+	return buckets, nil
+}
+
+// putobjectcallback
+func putobjectcallback(bucket, object string, filesize int64, metadata map[string]string) (result minio.ListObjectsInfo, err error) {
+	fmt.Printf("Go.putobjectcallback(): called \n")
```

Comment:
same as kaloyan's comment about fmt.Print*. better to use a logger

Reply:
Cleaned out the debug information